PatreonClient/ItemRelationshipAttribute.cs

```diff
@@ -0,0 +1,31 @@
+using System;
+
+namespace PatreonClient
+{
+    [AttributeUsage(AttributeTargets.Class, AllowMultiple = true)]
+    public class ItemRelationshipAttribute : Attribute
```

Comment:
Maybe it's worth making individual classes instead of this attribute
`User : PatreonData<UserAttributes, UserRelationships>`

Feels like this could contribute towards cleaning up generics across the solution.  What do you think?